src/bootstrap/config.rs

```diff
@@ -782,14 +771,75 @@ impl Config {
             set(&mut config.missing_tools, t.missing_tools);
         }
 
+        // This is duplicated in bootstrap.py.
+        config.out = build.build_dir.unwrap_or_else(|| PathBuf::from("build"));
+        t!(fs::create_dir_all(&config.out));
+        config.out = config.out.canonicalize().expect("Build directory canonicalized");
+        config.out = config
+            .out
+            .components()
+            .flat_map(|component| {
+                if component.as_os_str() == "$ROOT" {
+                    config.src.components().collect::<Vec<_>>()
+                } else {
+                    vec![component]
+                }
+            })
+            .collect();
+
+        if config.dry_run {
+            let dir = config.out.join("tmp-dry-run");
+            t!(fs::create_dir_all(&dir));
+            config.out = dir;
+        }
+
+        let build_rustc = build.rustc;
+        config.initial_rustc = env::var_os("RUSTC")
```

Comment:
Could this unconditionally be whatever rustc compiled the build script? (the build script should have a `RUSTC` env var)

Reply:
I think there's probably a cargo bug (or maybe rustup?), but for me that environment variable gets set to just "rustc" when compiling through rustup/cargo, even with `+beta`, which would mean that we don't use the right rustc. Not a problem with x.py, because that sets RUSTC in the environment which fixes things I believe.

I'm down for removing this whole section and trying to fix the upstream bug, wherever that is, though.

Reply:
Oh that's probably because Cargo just executes `rustc` itself, in that case could the build script read `RUSTC` and find it in `PATH`?